### ISSUE TITLE
Multiple Modules for the same `stationId` with different config, show the same data regardless of different `directionStationId`configuration

### ISSUE THREAD

**Issue by Timmeey** [deg0nz/MMM-PublicTransportBerlin#55]
### TLDR:
When having two modules for the same `stationID` but with different `directionStationID`s, it shows the same result for both modules.

### Expected behavior:
Each module uses its own config from the MM config file

### Actual behavior:
Modules seem to be using a random(or the last occurrence) config from the MM config file based on the same `StationID`

### Description
First of all, thank you for your work on this module, the stuff that works is really great.

I'm trying to achieve to separate the departures of the different directions, into different modules on the screen (so you can see at a glance when your S-Bahnhof is leaving, without checking the Destination (e.g. direction North, on the right widget, direction South the left widget). (The given example does it a little differently, but its about the DirectionStationId)

So I have two modules, one named `BornholmerSued`which should show all departures from Bonrholmer Str. going to Gesundbrunnen, and one module showing Departures from Bornholmer Str. going to Schoenhauser Allee. 

But what I'm getting is that it shows departures to Schoenhauser Allee only on both widgets. I already tried to switch the order of the modules in the config, even inserted some other modules in between the configs for both modules, but still the same result. Most of the time it seems to be using the LAST config for the station for both modules, but sometimes I think it used the second config, but both modules always show the same data.

I tried the same thing with some Tram stations, and got the same result. 
If I have 4 modules, 2 for `StationA` and 2 for `StationB`, both `StationA`modules show the same data and both `StationB`modules show the same data. SO I CAN. have multiple modules but only if they are about different Stations. 
It feels like it only fetches the data ONCE per `StationID`and then uses that data for all modules requesting the same `StationID`, but without taking into account the different other parameters/filters 

### Example
Version: 6cfcbeaaad592ee4025ea48ffd9c1f4ace295652

Config:
```
{
            module: "MMM-PublicTransportBerlin",
            position: "right",
            config: {
                name: "BornholmerRing",
                stationId: "900000110011",
                hidden: false,
                //Schoenhauser Allee
                directionStationId: "900000110001",
                ignoredLines: [],
                excludedTransportationTypes: "bus, tram",   
                travelTimeToStation: 10,
                interval: 120001,
                departureMinutes: 25,          
                maxDepartures: 15,
                marqueeLongDirections: true,
                showColoredLineSymbols: true,  
                useColorForRealtimeInfo: true,
                showTableHeaders: true,
                showTableHeadersAsSymbols: true,
                maxUnreachableDepartures: 3,    
                maxReachableDepartures: 6,
                fadeUnreachableDepartures: true,
                fadeReachableDepartures: true,
                fadePointForReachableDepartures: 0.25,
                excludeDelayFromTimeLabel: true
            }
        },
        {
            module: "MMM-PublicTransportBerlin",
            position: "left",
            config: {
                name: "BornholmerSued",
                stationId: "900000110011",
                hidden: false,
                //Gesundbrunnen
                directionStationId: "900000007102",
                ignoredLines: [],
                excludedTransportationTypes: "bus, tram",   
                travelTimeToStation: 10,
                interval: 120000,
                departureMinutes: 25,          
                maxDepartures: 15,
                marqueeLongDirections: true,
                showColoredLineSymbols: true,  
                useColorForRealtimeInfo: true,
                showTableHeaders: true,
                showTableHeadersAsSymbols: true,
                maxUnreachableDepartures: 3,    
                maxReachableDepartures: 6,
                fadeUnreachableDepartures: true,
                fadeReachableDepartures: true,
                fadePointForReachableDepartures: 0.25,
                excludeDelayFromTimeLabel: true
            }
        },
```

![Bildschirmfoto 2020-09-27 um 17 50 41](https://user-images.githubusercontent.com/1174542/94369489-41ba2e80-00ea-11eb-9a13-5ab5acc78ebf.png)

**Comment by deg0nz** [comment]
Thanks for the detailed error report! This is definitely a bug within the MM-Module. 

Since module instances are identified by the stationId, both instances get the same data, because the listen to the same data trigger. The correct data for the second instance will be downloaded to the mirror, but both instances react to the same answer, so the data is always the same. I've never tested the modules that way. Good that you found it 👍 

I'm trying to fix this tonight. I will notify you when it's done.

**Comment by Timmeey** [comment]
I'm by no means a JS/Node Dev. But from what I can see, the fetchers are kept in a map, while the key to the fetcher is the station ID.

For my particular case, using the given `config.Name` would help. But this might not be backwards compatible for other users, which are already using multiple modules, but did not give them a name. 

So my totally uninformed suggestion to make a backwards compatible change, is to use a compound key. Consisting of `config.name-config.stationID` while `config.Name` being optional. 

That way the behavior for existing configs would not change (assuming no one is actively using this bug-behavior), while still fixing the issue.

**Comment by Timmeey** [comment]
Thanks for the change. But this will break older setups, that did not provide a name, as the name was not a required parameter.

**Comment by deg0nz** [comment]
Oh that's crazy. I didn't know, that mentioning "fix" and the issue is closing it :)

The issue should be fixed now. Please open it, if you still encounter issues @Timmeey.

With commit 22947de2498518e546b4359666efb42077cad765 you are also able to configure if you want to show directions in the module's header.

**Comment by deg0nz** [comment]
> Thanks for the change. But this will break older setups, that did not provide a name, as the name was not a required parameter.

Yeah, I'm aware of that. We will see, if this breaks a lot of stuff. After some thinking, this was the best option to not unnecessarily increase code complexity.

And since it was already mentioned that this value should be set, we have the prerequisites to do it that way.

**Comment by deg0nz** [comment]
I added a Fallback options for generating the module name on startup, if it is empty or points to default.
With commit b03eaae0c6c49716bf2cec5eced543175cc761d3 all the changed should be backwards compatible.
